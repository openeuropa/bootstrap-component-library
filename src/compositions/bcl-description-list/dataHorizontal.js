module.exports = {
  variant: "horizontal",
  items: [
    {
      term: {
        label: "Only title label with icon",
        icon: {
          name: "geo-alt-fill",
          size: "xs",
          path: "/icons.svg",
        },
      },
      definition: "Description text goes here.",
    },
    {
      term: [
        {
          label: "First title label",
        },
        {
          label: "Second title label",
          icon: {
            name: "geo-alt-fill",
            size: "xs",
            path: "/icons.svg",
          },
        },
      ],
      definition: "text goes here.",
    },
    {
      term: {
        label: "Only title label",
      },
      definition: [
        {
          label: "Description text goes here.",
        },
        {
          label: "Description text goes here.",
        },
      ],
    },
  ],
};
