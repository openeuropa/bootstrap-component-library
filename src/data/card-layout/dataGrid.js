const drupalAttribute = require("drupal-attribute");

module.exports = {
  type: "grid",
  responsive_columns: 3,
  responsiveness: "md",
  gutter: 2,
  items: [
    {
      title: {
        content: "Title card 1",
      },
      subtitle: {
        content: "Subtitle card 1",
        classes: "mb-2",
      },
      text: {
        content: "Text card with longer text which is supported 1",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1002/600/400",
        alt: "alt img",
        position: "top",
      },
      card_header: "Header of card 1",
      card_footer: "Footer of card 1",
      wrapper_class: "col",
      content: "<button class='btn btn-primary'>Custom action</button>",
      attributes: new drupalAttribute().addClass("mb-3"),
    },
    {
      title: {
        content: "Title card 2",
      },
      subtitle: {
        content: "Subtitle card 2",
        classes: "mb-2",
      },
      text: {
        content: "Text card with longer text which is supported 2",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1003/600/400",
        alt: "alt img",
        position: "top",
      },
      card_header: "Header of card 2",
      card_footer: "Footer of card 2",
      wrapper_class: "col",
      attributes: new drupalAttribute().addClass("mb-3"),
      content:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    },
    {
      title: {
        content: "Title card 3",
      },
      subtitle: {
        content: "Subtitle card 3",
        classes: "mb-2",
      },
      text: {
        content:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1004/600/400",
        alt: "alt img",
        position: "top",
      },
      card_header: "Header of card 3",
      card_footer: "Footer of card 3",
      wrapper_class: "col",
      content: "<p>Custom text</p>",
      attributes: new drupalAttribute().addClass("mb-3"),
    },
    {
      title: {
        content: "Title card 4",
      },
      subtitle: {
        content: "Subtitle card 4",
        classes: "mb-2",
      },
      text: {
        content: "Text card with longer text which is supported 4",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1005/600/400",
        alt: "alt img",
        position: "top",
      },
      card_header: "Header of card 4",
      card_footer: "Footer of card 4",
      wrapper_class: "col",
      attributes: new drupalAttribute().addClass("mb-3"),
      content:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    },
    {
      title: {
        content: "Title card 5",
      },
      subtitle: {
        content: "Subtitle card 5",
        classes: "mb-2",
      },
      text: {
        content: "Text card with longer text which is supported 5",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1006/600/400",
        alt: "alt img",
        position: "top",
      },
      card_header: "Header of card 5",
      card_footer: "Footer of card 5",
      wrapper_class: "col",
      attributes: new drupalAttribute().addClass("mb-3"),
    },
    {
      title: {
        content: "Title card 6",
      },
      subtitle: {
        content: "Subtitle card 6",
        classes: "mb-2",
      },
      text: {
        content: "Text card with longer text which is supported 6",
        classes: "mb-2",
      },
      image: {
        path: "https://picsum.photos/seed/1007/600/400",
        alt: "alt img",
        position: "bottom",
      },
      card_header: "Header of card 6",
      card_footer: "Footer of card 6",
      wrapper_class: "col",
      attributes: new drupalAttribute().addClass("mb-3"),
      content:
        "<p>Some representative placeholder content for the first card.</p>",
    },
  ],
};
