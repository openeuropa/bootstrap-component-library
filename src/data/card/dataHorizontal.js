const drupalAttribute = require("drupal-attribute");

module.exports = {
  horizontal: true,
  horizontal_grid: {
    left_col: 4,
    right_col: 8,
    gutter: 0,
  },
  title: {
    content: "Title card",
  },
  subtitle: {
    content: "Subtitle card",
    classes: "mb-2",
  },
  text: {
    content: "Text card with longer text which is supported",
    classes: "mb-2",
  },
  image: {
    path: "https://picsum.photos/seed/1019/400/600",
    alt: "alt img",
    position: "top",
  },
  content: "<button class='btn btn-primary'>Custom action</button>",
  attributes: new drupalAttribute().addClass("w-50"),
};
