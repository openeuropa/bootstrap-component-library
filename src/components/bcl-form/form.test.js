import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import dataBasic from "@openeuropa/bcl-data-form/data";
import dataGrid from "@openeuropa/bcl-data-form/data--grid";
import dataDisabled from "@openeuropa/bcl-data-form/data--disabled";
import dataInline from "@openeuropa/bcl-data-form/data--inline";
import dataHorizontal from "@openeuropa/bcl-data-form/data--horizontal";

const template = "@oe-bcl/bcl-form/form.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Form", () => {
  test(`basic renders correctly`, () => {
    expect.assertions(1);

    return expect(render(dataBasic)).resolves.toMatchSnapshot();
  });

  test(`horizontal renders correctly`, () => {
    expect.assertions(1);

    return expect(render(dataHorizontal)).resolves.toMatchSnapshot();
  });

  test(`using a grid renders correctly`, () => {
    expect.assertions(1);

    return expect(render(dataGrid)).resolves.toMatchSnapshot();
  });

  test(`renders correctly with title`, () => {
    expect.assertions(1);

    return expect(
      render({
        ...dataBasic,
        title: "Carousel Title",
        title_tag: "h1",
        title_link: {
          path: "/example.html",
          label: "A title created by the bcl heading template",
        },
      })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly when disabled`, () => {
    expect.assertions(1);

    return expect(render(dataDisabled)).resolves.toMatchSnapshot();
  });

  test(`inline renders correctly`, () => {
    expect.assertions(1);

    return expect(render(dataInline)).resolves.toMatchSnapshot();
  });
});
