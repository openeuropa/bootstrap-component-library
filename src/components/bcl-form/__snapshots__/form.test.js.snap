// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Form basic renders correctly 1`] = `
<jest>
  <form
    class="needs-validation"
    novalidate="true"
    onsubmit="return false;"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleInputEmail1"
      >
        Email address *
      </label>
      <input
        aria-describedby="emailHelp"
        class="form-control"
        id="exampleInputEmail1"
        required="true"
        type="email"
      />
      <div
        class="invalid-feedback"
      >
        Please provide an email.
      </div>
      <div
        class="valid-feedback"
      >
        Valid feedback regarding email.
      </div>
      <div
        class="form-text"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </div>
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleInputPassword1"
      >
        Password *
      </label>
      <input
        class="form-control"
        id="exampleInputPassword1"
        required="true"
        type="password"
      />
      <div
        class="invalid-feedback"
      >
        Please provide a password.
      </div>
      <div
        class="valid-feedback"
      >
        Valid feedback regarding password.
      </div>
    </div>
    <div
      class="mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="exampleInputPassword1"
          required="true"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="exampleInputPassword1"
        >
          Check me out *
        </label>
        <div
          class="invalid-feedback"
        >
          Please check the required checkbox.
        </div>
        <div
          class="valid-feedback"
        >
          Valid feedback regarding check.
        </div>
      </div>
    </div>
    <button
      class="btn btn-primary btn-md"
      type="submit"
    >
      Submit
    </button>
  </form>
</jest>
`;

exports[`OE - Form horizontal renders correctly 1`] = `
<jest>
  <form
    onsubmit="return false;"
  >
    <div
      class="row mb-3"
    >
      <label
        class="col-form-label col-sm-2"
        for="exampleInputEmail1"
      >
        Email address
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          id="exampleInputEmail1"
          required="true"
          type="email"
        />
      </div>
    </div>
    <div
      class="row mb-3"
    >
      <label
        class="col-form-label col-sm-2"
        for="exampleInputPassword1"
      >
        Password
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          id="exampleInputPassword1"
          required="true"
          type="password"
        />
      </div>
    </div>
    <fieldset
      class="row mb-3"
    >
      <legend
        class="col-form-label col-sm-2 pt-0"
      >
        Radios
      </legend>
      <div
        class="col-sm-10"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="exampleInputRadios1"
            type="radio"
          />
          <label
            class="form-check-label"
            for="exampleInputRadios1"
          >
            First radio
          </label>
        </div>
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="exampleInputRadios2"
            type="radio"
          />
          <label
            class="form-check-label"
            for="exampleInputRadios2"
          >
            Second radio
          </label>
        </div>
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            disabled="true"
            id="exampleInputRadios3"
            type="radio"
          />
          <label
            class="form-check-label"
            for="exampleInputRadios3"
          >
            Third radio
          </label>
        </div>
      </div>
    </fieldset>
    <div
      class="row mb-3"
    >
      <div
        class="col-sm-10 offset-sm-2"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="exampleInputCheck1"
            required="true"
            type="checkbox"
          />
          <label
            class="form-check-label"
            for="exampleInputCheck1"
          >
            Example checkbox
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-12"
    >
      <button
        class="btn btn-primary btn-md"
        type="submit"
      >
        Sign in
      </button>
    </div>
  </form>
</jest>
`;

exports[`OE - Form inline renders correctly 1`] = `
<jest>
  <form
    class="row row-cols-lg-auto g-3 align-items-center"
    novalidate="true"
    onsubmit="return false;"
  >
    <div
      class="col-12"
    >
      <label
        class="form-label visually-hidden"
        for="inlineFormInputGroupUsername"
      >
        Username
      </label>
      <input
        class="form-control"
        id="inlineFormInputGroupUsername"
        placeholder="Username"
        required="true"
        type="text"
      />
    </div>
    <div
      class="col-12"
    >
      <label
        class="form-label visually-hidden"
      >
        Preference
      </label>
      <select
        class="form-select"
        required="true"
      >
        <option
          selected=""
          value=""
        >
          Choose
        </option>
        <option
          value="1"
        >
          One
        </option>
        <option
          value="2"
        >
          Two
        </option>
        <option
          value="3"
        >
          Three
        </option>
      </select>
    </div>
    <div
      class="col-12"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="exampleInputCheck1"
          required="true"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="exampleInputCheck1"
        >
          Remember me *
        </label>
      </div>
    </div>
    <div
      class="col-12"
    >
      <button
        class="btn btn-primary btn-md"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</jest>
`;

exports[`OE - Form renders correctly when disabled 1`] = `
<jest>
  <form
    novalidate="true"
    onsubmit="return false;"
  >
    <fieldset
      disabled=""
    >
      <legend>
        Disabled fieldset example
      </legend>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="disabledTextInput"
        >
          Disabled input
        </label>
        <input
          aria-describedby="helperText"
          class="form-control"
          id="disabledTextInput"
          placeholder="Disabled input"
          required="true"
          type="text"
        />
        <div
          class="form-text"
          id="helperText"
        >
          Helper text for the form input
        </div>
      </div>
      <div
        class="mb-3"
      >
        <select
          class="form-select"
        >
          <option
            value="1"
          >
            a select option
          </option>
          <option
            value="2"
          >
            another select option
          </option>
          <option
            value="3"
          >
            another option
          </option>
          <option
            value="4"
          >
            last option
          </option>
          <option
            selected=""
            value=""
          >
            the selected option
          </option>
        </select>
        <div
          class="form-text"
          id="helperText"
        >
          Helper text for the select element
        </div>
      </div>
      <div
        class="mb-3"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="disabledFieldsetCheck"
            type="checkbox"
          />
          <label
            class="form-check-label"
            for="disabledFieldsetCheck"
          >
            Can't check this
          </label>
        </div>
      </div>
    </fieldset>
    <button
      class="btn btn-primary btn-md"
      type="submit"
    >
      Submit
    </button>
  </form>
</jest>
`;

exports[`OE - Form using a grid renders correctly 1`] = `
<jest>
  <form
    class="row g-3"
    novalidate="true"
    onsubmit="return false;"
  >
    <div
      class="col-md-6"
    >
      <label
        class="form-label"
        for="exampleInputEmail1"
      >
        Email address *
      </label>
      <input
        aria-describedby="emailHelp"
        class="form-control"
        id="exampleInputEmail1"
        required="true"
        type="email"
      />
      <div
        class="form-text"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </div>
    </div>
    <div
      class="col-md-6"
    >
      <label
        class="form-label"
        for="exampleInputPassword1"
      >
        Password *
      </label>
      <input
        class="form-control"
        id="exampleInputPassword1"
        required="true"
        type="password"
      />
    </div>
    <div
      class="col-12"
    >
      <label
        class="form-label"
        for="exampleInputAddress1"
      >
        Address
      </label>
      <input
        class="form-control"
        id="exampleInputAddress1"
        type="text"
      />
    </div>
    <div
      class="col-12"
    >
      <label
        class="form-label"
        for="exampleInputAddress2"
      >
        Address 2
      </label>
      <input
        class="form-control"
        id="exampleInputAddress2"
        type="text"
      />
    </div>
    <div
      class="col-md-6"
    >
      <label
        class="form-label"
        for="exampleInputCity1"
      >
        City *
      </label>
      <input
        class="form-control"
        id="exampleInputCity1"
        required="true"
        type="text"
      />
    </div>
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
        for="exampleInputState1"
      >
        State *
      </label>
      <input
        class="form-control"
        id="exampleInputState1"
        required="true"
        type="text"
      />
    </div>
    <div
      class="col-md-2"
    >
      <label
        class="form-label"
        for="exampleInputZip1"
      >
        Zip *
      </label>
      <input
        class="form-control"
        id="exampleInputZip1"
        required="true"
        type="text"
      />
    </div>
    <div
      class="col-12"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="exampleInputCheck1"
          required="true"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="exampleInputCheck1"
        >
          Check me out *
        </label>
      </div>
    </div>
    <div
      class="col-12"
    >
      <button
        class="btn btn-primary btn-md"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</jest>
`;
